examples/browser-mfs/package.json

```diff
@@ -13,6 +13,7 @@
   "author": "",
   "license": "ISC",
   "devDependencies": {
+    "@babel/plugin-transform-runtime": "^7.10.1",
```

Comment:
I was getting `regeneratorRuntime is not defined` from the browser bundle from `libp2p-noise`. This solves the issue here. Is this soemthig that we want here?

The other examples did not have this problems, but they also have different webpack/other configurations. Is there any other place that we should document this?

Reply:
This should not be needed, what is libp2p-noise doing to force this?

Reply:
Thanks, I will open an issue on `libp2p-noise`

Reply:
If they we aegir they can avoid all the tooling besides typescript.
So if they run ts to just compile to js into a src folder they can run aegir the same way all others repos do.

Reply:
> libp2p-noise is running babel in published code they **really** should not!

We use babel to transpile from ts to js so we **really** have to run it. Seems like my targets in browserlist in babel includes browser that doesn't support async generator so babel relied on global `regeneratorRuntime`. Ref: https://github.com/babel/babel/issues/9849#issuecomment-487040428

I will disable this, which will probably result in slightly larger generated codesize but it should work without hacks here.

I would really like to use aegir but I don't wan't to generate ts into src directory, because I would have to commit src directory with generated code to satisfy aegir. We will integrate aegir once there is first class support for typescript codebase. Currently from src (typescript) we generate using babel `build` directory with node optimized code and `lib` directory with esmodules to enable tree shaking in web apps.

Reply:
> You really shouldn't run babel, you are making decisions for your users and publishing non optimal code. Only the users know their targets.

[Aegir is doing same thing](https://github.com/ipfs/aegir/blob/master/src/config/webpack.config.js#L41), it's using babel to target some browsers and nodejs and then using webpack creating umd build that can run in both. I generally don't like to do this because it's replacing some js features with ugly (and slower) hacks to accomodate browsers which doesn't support that (async generators for example).

> IMO you should run TS to compile to JS without any transformation except types obviously. You can still tell TS to output esm and cjs. If your only concern about esm is tree shaking you probably can just skip it, i doubt it makes any difference.

That won't work in nodejs since it doesn't support ECMAScript Modules

> Regarding aegir, it doesn't care if the code is committed or not. The only problem that I see is that you are already using a folder called src for TS and that's kinda hard coded in aegir. I can make it configurable for you, that should solve your problem.

If you enable configurable source we will enable aegir and switch to umd build.

FYI I've set aegir babel browsers targets in libp2p noise and it seems to pass all example tests locally: https://github.com/NodeFactoryIo/js-libp2p-noise/pull/62

Reply:
> Aegir is doing same thing, it's using babel to target some browsers and nodejs and then using webpack creating umd build that can run in both. I generally don't like to do this because it's replacing some js features with ugly (and slower) hacks to accomodate browsers which doesn't support that (async generators for example).

Yes aegir runs babel with browserlist, but this is only because we publish a  special browser bundle to be used with script tags directly using js cdns like unpkg or jsdelivr, NOT to be bundled again by users!
The `main` prop in package.json always points to `src/index.js`

>  That won't work in nodejs since it doesn't support ECMAScript Modules

Yes, thats why i said to run TS twice one for esm and another for cjs, package.json `main` would point to `src-cjs/index.js` and `module` to `src-esm/index.js`. Or just have cjs cause esm in your case likely adds 0 value.

> If you enable configurable source we will enable aegir and switch to umd build.

I will make this happen asap.